File: Source/iOS/Genealogy/Genealogy/Model/A04/DTO/SettingInfoDTO.m
//
//  SettingInfoDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "SettingInfoDTO.h"

@implementation SettingInfoDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"value1"] isKindOfClass:[NSNull class]]) {
        self.value1 = jsonData[@"value1"];
    }
    if (![jsonData[@"value2"] isKindOfClass:[NSNull class]]) {
        self.value2 = jsonData[@"value2"];
    }
    if (![jsonData[@"value3"] isKindOfClass:[NSNull class]]) {
        self.value3 = jsonData[@"value3"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.value1) {
        dic[@"value1"] = self.value1;
    }
    if (self.value2) {
        dic[@"value2"] = self.value2;
    }
    if (self.value3) {
        dic[@"value3"] = self.value3;
    }
    return dic;
}

@end

File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a04/dto/SettingInfoDTO.java
/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a04.dto;

/**
 * SettingInfoDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/14             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("家族足迹数据")
public class SettingInfoDTO implements java.io.Serializable, Comparable<SettingInfoDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private Boolean value1 = null;

	public Boolean getValue1() {
		return this.value1;
	}

	public void setValue1(
			Boolean value1) {
		this.value1 = value1;
	}

	private Boolean value2 = null;

	public Boolean getValue2() {
		return this.value2;
	}

	public void setValue2(
			Boolean value2) {
		this.value2 = value2;
	}

	private Boolean value3 = null;

	public Boolean getValue3() {
		return this.value3;
	}

	public void setValue3(
			Boolean value3) {
		this.value3 = value3;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		SettingInfoDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}
